src/Compilers/Core/CodeAnalysisTest/AnalyzerConfigTests.cs

```diff
@@ -1455,6 +1455,40 @@ public void DiagnosticIdInstancesAreSharedBetweenMultipleTrees()
                     options[2].TreeOptions.Keys.First()));
         }
 
+        [Fact]
+        public void TreesShareOptionsInstances()
+        {
+            var configs = ArrayBuilder<AnalyzerConfig>.GetInstance();
+            configs.Add(Parse(@"
+[*.cs]
+dotnet_diagnostic.cs000.severity = warning", "/.editorconfig"));
+
+            var options = GetAnalyzerConfigOptions(
+                new[] { "/a.cs", "/b.cs", "/c.cs" },
+                configs);
+            configs.Free();
+
+            Assert.True(
```

Comment:
consider using `Assert.Same`

Reply:
I think this is clearer.

Reply:
I disagree. `Assert.Same` exists for this reason, and will give a better error message if it fails.